input-output-hk/cardano-sl
# [CDEC-509] Remove HasCoreConfiguration and HasConfiguration

## Description

This is the removal of the final `Data.Reflection` constraint in `core`! This PR removes the `dbSerializeVersion` configuration parameter, a so far unused value. Could reviewers please check that a node running the code before this commit **continues to sync after this commit is applied**.



## Linked issue

[CDEC-509](https://iohk.myjetbrains.com/youtrack/issue/CDEC-509)



## Type of change

- [~] 🐞 Bug fix (non-breaking change which fixes an issue)
- [~] 🛠 New feature (non-breaking change which adds functionality)
- [~] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [~] 🔨 New or improved tests for existing code
- [~] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [~] CHANGELOG entry has been added and is linked to the correct PR on GitHub.

## Testing checklist

- [~] I have added tests to cover my changes.

## QA Steps


1) Connect to the testnet running a node on `develop` and sync some blocks
2) Kill the node and then with **the same state** connect with a node from this PR
3) Verify that the DB is still read and the syncing continues

## Screenshots (if available)
